app/pages/project/workflow-selection.jsx

```diff
@@ -97,14 +98,19 @@ class WorkflowSelection extends React.Component {
         this.setState({ error, loadingSelectedWorkflow: false });
       }
     })
-    .then(([workflow]) => {
-      if (workflow) {
+    .then((workflow) => {
+      const workflowMatchesProjectID = this.idsMatch(workflow.links.project, this.props.project.id)
+      const allowedActiveWorkflow = activeFilter && workflow && workflow.active
+      const ignoreActiveFilter = !activeFilter && workflow
+      const loadableWorkflow = allowedActiveWorkflow || ignoreActiveFilter
+
+      if (workflowMatchesProjectID && loadableWorkflow) {
```

Comment:
I'm not 100% sure about the consequence of not loading if the project doesn't match. From my understanding this will call itself again and try to pick a workflow that will match. It may get stuck in a loop as this happened when i was testing this, e.g. set the `workflowMatchesProjectID = false` but this is contrived and i expect that at least 1 workflow will be loaded correctly.

Reply:
I'm not able to get workflows to load, since this change was committed.